Source/PublicAPI/Encryption/EThree+Ratchet.swift

```diff
@@ -0,0 +1,139 @@
+//
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     (1) Redistributions of source code must retain the above copyright
+//     notice, this list of conditions and the following disclaimer.
+//
+//     (2) Redistributions in binary form must reproduce the above copyright
+//     notice, this list of conditions and the following disclaimer in
+//     the documentation and/or other materials provided with the
+//     distribution.
+//
+//     (3) Neither the name of the copyright holder nor the names of its
+//     contributors may be used to endorse or promote products derived from
+//     this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE AUTHOR ''AS IS'' AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+// WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT,
+// INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+// Lead Maintainer: Virgil Security Inc. <support@virgilsecurity.com>
+//
+
+import VirgilSDK
+import VirgilSDKRatchet
+import VirgilCryptoRatchet
+
+public extension EThree {
+    func createRatchetChat(with card: Card, name: String? = nil) -> GenericOperation<RatchetChat> {
+        return CallbackOperation { _, completion in
+            do {
+                let secureChat = try self.getSecureChat()
+
+                guard secureChat.existingSession(withParticipantIdentity: card.identity, name: name) == nil else {
+                    throw EThreeRatchetError.chatAlreadyExists
+                }
+
+                guard card.identity != self.identity else {
+                    throw EThreeRatchetError.selfChatIsForbidden
+                }
+
+                let session = try secureChat.startNewSessionAsSender(receiverCard: card, name: name)
+                    .startSync()
+                    .get()
+
+                let ticket = try session.encrypt(string: UUID().uuidString)
+
+                try self.cloudRatchetStorage.store(ticket, sharedWith: card, name: name)
+
+                try secureChat.storeSession(session)
+
+                let ratchetChat = RatchetChat(session: session,
+                                              sessionStorage: secureChat.sessionStorage)
+
+                completion(ratchetChat, nil)
+            }
+            catch let error as NSError where error.code == 50_017 {
```

Comment:
What's that?

Reply:
And why NSError? We don't use NSError

Reply:
That's server error from ratchet, which tells there's no keys of specified user.

Reply:
I can use [ServiceError](https://github.com/VirgilSecurity/virgil-sdk-x/blob/master/Source/BaseClient/ServiceError.swift) from core sdk, but it still inherits from NSError

Reply:
If ServiceError is thrown, then ServiceError should be caught. Which method throws this error in this scope?

Reply:
SecureChat.startNewSessionAsSender throws

Reply:
Error should be more descriptive. If user has card he IS registered.

Reply:
Now catches ServiceError. Changed error name